src/WireMock.Net/ResponseBuilders/Response.cs

```diff
@@ -264,34 +259,19 @@ public IResponseBuilder WithProxy(string proxyUrl)
         /// With Proxy URL.
         /// </summary>
         /// <param name="proxyUrl">The proxy url.</param>
-        /// <param name="clientX509Certificate2Filename">The X509Certificate2 file to use for client authentication.</param>
+        /// <param name="clientX509Certificate2ThumbprintOrSubjectName">The X509Certificate2 file to use for client authentication.</param>
         /// <returns>A <see cref="IResponseBuilder"/>.</returns>
-        public IResponseBuilder WithProxy(string proxyUrl, string clientX509Certificate2Filename)
+        public IResponseBuilder WithProxy(string proxyUrl, string clientX509Certificate2ThumbprintOrSubjectName)
```

Comment:
You could add:
`[NotNull] string proxyUrl`
and 
`[NotNull] clientX509Certificate2ThumbprintOrSubjectName`